core/src/test/java/software/amazon/awssdk/core/pagination/PaginatedItemsIterableTest.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2010-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.core.pagination;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.Iterator;
+import java.util.function.Function;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+public class PaginatedItemsIterableTest {
+
+    private SdkIterable pagesIterable = Mockito.mock(SdkIterable.class);
```

Comment:
If you do the RunWith mockito runner you can annotate these with just Mock

Reply:
Oh thats it. I tried with @Mock annotations but I was getting NPE. I forgot to annotate the class with MockitoJUnitRunner.